docs/reference/rocmcc/rocmcc.md

```diff
@@ -3,21 +3,892 @@
+ROCm currently provides two compiler interfaces for compiling HIP programs:
+- /opt/rocm/bin/hipcc
+- /opt/rocm/bin/amdclang++
+
+Both leverage the same LLVM compiler technology with the AMD GCN GPU support; however, they offer a slightly different user experience. The hipcc command-line interface aims to provide a more familiar user interface to users who are experienced in CUDA but relatively new to the ROCm/HIP development environment. On the other hand, amdclang++ provides a user interface identical to the clang++ compiler. It is more suitable for experienced developers who want to directly interact with the clang compiler and gain full control of their application’s build process.
+
+The major differences between hipcc and amdclang++ are listed below:
+||
+|:--:|
+| <b>Table 3. Differences Between hipcc and amdclang++</b>|
+||
+
+|| Hipcc | amdclang++ |
+| ----------- | ----------- | ----------- |
+| Compiling HIP source files | Treats all source files as HIP language source files | Enables the HIP language support for files with the “.hip” extension or through the -x hip compiler option |
+| Automatic GPU architecture detection | Auto-detects the GPUs available on the system and generates code for those devices when no GPU architecture is specified | Has AMD GCN gfx803 as the default GPU architecture. The --offload-arch compiler option may be used to target other GPU architectures |
+| Finding a HIP installation | Finds the HIP installation based on its own location and its knowledge about the ROCm directory structure | First looks for HIP under the same parent directory as its own LLVM directory and then falls back on /opt/rocm. Users can use the --rocm-path option to instruct the compiler to use HIP from the specified ROCm installation. |
+| Linking to the HIP runtime library | Is configured to automatically link to the HIP runtime from the detected HIP installation | Requires the --hip-link flag to be specified to link to the HIP runtime. Alternatively, users can use the -l\<dir\> -lamdhip64 option to link to a HIP runtime library. |
+| Device function inlining | Inlines all GPU device functions, which provide greater performance and compatibility for codes that contain file scoped or device function scoped \_\_shared\_\_ variables. However, it may increase compile time. | Relies on inlining heuristics to control inlining. Users experiencing performance or compilation issues with code using file scoped or device function scoped \_\_shared\_\_ variables could try -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false to work around the issue. There are plans to address these issues with future compiler improvements. |
+| Source code location | Developed at [https://github.com/ROCm-Developer-Tools/HIPCC](https://github.com/ROCm-Developer-Tools/HIPCC) | Developed at [https://github.com/RadeonOpenCompute/llvm-project](https://github.com/RadeonOpenCompute/llvm-project) |
+
+
+
+# Compiler Options and Features
+
```

Comment:
Please do not use Level 1 headings except for the page title

Reply:
Fixed, moved all titles one step down the hierarchy

Reply:
Please note that rocmcc is still a work-in-progress draft, still working on a few missing parts and cleaning table 9. Thanks